src/main/java/core/basesyntax/Circle.java

```diff
@@ -0,0 +1,35 @@
+package core.basesyntax;
+
+public class Circle extends Figure {

```

Comment:
Don't extend your `Main/Application` class from `FigureSupplier` or `ColorSupplier`. It seems like `Circle` is extending `Figure`, which is fine if `Figure` is not an abstract class that sets behavior for classes. Ensure that `Figure` is either an interface with default methods or a concrete class that does not enforce behavior.